packages/toolkit/src/components/card-pipeline-hub/Head.tsx

```diff
@@ -0,0 +1,136 @@
+"use client";
+
+import * as React from "react";
+import {
+  InstillStore,
+  OrganizationOwner,
+  Pipeline,
+  UserOwner,
+  toastInstillError,
+  useDeleteUserPipeline,
+  useInstillStore,
+  useShallow,
+} from "../../lib";
+import { Icons, useToast } from "@instill-ai/design-system";
+import { EntityAvatar } from "../EntityAvatar";
+import { useRouter } from "next/navigation";
+
+export const HeadSkeleton = () => {
+  return (
+    <div className="flex w-full flex-row gap-x-2 p-3">
+      <div className="my-auto h-8 w-8 shrink-0 grow-0 animate-pulse rounded-full bg-semantic-bg-secondary" />
+      <div className="h-[38px] w-20 animate-pulse rounded bg-semantic-bg-secondary"></div>
+    </div>
+  );
+};
+
+const selector = (store: InstillStore) => ({
+  accessToken: store.accessToken,
+  enabledQuery: store.enabledQuery,
+});
+
+export const Head = ({
+  ownerID,
+  isOwner,
+  pipeline,
+}: {
+  ownerID: string;
+  isOwner: boolean;
+  pipeline: Pipeline;
+}) => {
+  const router = useRouter();
+  const { toast } = useToast();
+  const { accessToken } = useInstillStore(useShallow(selector));
+
+  const deletePipeline = useDeleteUserPipeline();
+  async function handleDeletePipeline() {
+    try {
+      await deletePipeline.mutateAsync({
+        pipelineName: pipeline.name,
+        accessToken: accessToken ? accessToken : null,
+      });
+
+      toast({
+        title: "Pipeline deleted",
+        variant: "alert-success",
+        size: "large",
+      });
+    } catch (error) {
+      toastInstillError({
+        title: "Something went wrong when delete the pipeline",
+        error,
+        toast,
+      });
+    }
+  }
+
+  const pipelineAvatar = React.useMemo(() => {
+    if (pipeline.owner_name.split("/")[0] === "users") {
+      return (pipeline.owner as UserOwner).user.profile?.avatar ?? null;
+    }
+
+    if (pipeline.owner_name.split("/")[0] === "organizations") {
+      return (
+        (pipeline.owner as OrganizationOwner).organization.profile?.avatar ??
+        null
+      );
+    }
+
+    return null;
+  }, [pipeline]);
+
+  const displayName = React.useMemo(() => {
+    const owner = pipeline.owner as UserOwner;
+    const username = owner?.user?.name?.split("/")[1]; // Get the part after "users/"
+    const ownerName =
+      owner?.user?.profile?.company_name ||
+      owner?.user?.profile?.display_name ||
+      "";
```

Comment:
@thewbuk In our product, due to we follow the Google AIP resource oriented api design. All the `name` have very unique meaning, which means the full path toward the resource like `users/uid`. So I will recommend change the naming here

- `username` -> `userID`
- `ownername` -> `ownerDisplayName`

Please don't be afraid to have long variable name in this repo, if we can make the meaning clear, I think that is a good practice